tests/test_cli.py

```diff
@@ -185,18 +198,23 @@ def test_find_files_recursively(self):
                                      '  - \'*\'\n')
         self.assertEqual(
             sorted(cli.find_files_recursively([self.wd], conf)),
-            [os.path.join(self.wd, 'a.yaml'),
-             os.path.join(self.wd, 'c.yaml'),
-             os.path.join(self.wd, 'dos.yml'),
-             os.path.join(self.wd, 'empty.yml'),
-             os.path.join(self.wd, 'en.yaml'),
-             os.path.join(self.wd, 'no-yaml.json'),
-             os.path.join(self.wd, 'non-ascii/éçäγλνπ¥/utf-8'),
-             os.path.join(self.wd, 's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'),
-             os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
-             os.path.join(self.wd, 'sub/directory.yaml/not-yaml.txt'),
-             os.path.join(self.wd, 'sub/ok.yaml'),
-             os.path.join(self.wd, 'warn.yaml')]
+            sorted(
+                [os.path.join(self.wd, 'a.yaml'),
+                 os.path.join(self.wd, 'c.yaml'),
+                 os.path.join(self.wd, 'dos.yml'),
+                 os.path.join(self.wd, 'empty.yml'),
+                 os.path.join(self.wd, 'en.yaml'),
+                 os.path.join(self.wd, 'no-yaml.json'),
+                 os.path.join(self.wd,
+                              's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'),
+                 os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
+                 os.path.join(self.wd, 'sub/directory.yaml/not-yaml.txt'),
+                 os.path.join(self.wd, 'sub/ok.yaml'),
+                 os.path.join(self.wd, 'warn.yaml')]
+                +
+                ([] if self.utf8_missing else
```

Comment:
hard to get this part looking good and passing the linter as well. any suggestions welcome.